packages/protoplugin/src/create-es-plugin.ts

```
@@ -202,7 +201,12 @@ export function createEcmaScriptPlugin(init: PluginInit): Plugin {
       // but no generate function was provided.  This also means that we will
       // have generated .ts files above.
       if (transpileJs || transpileDts) {
-        const transpileFn = init.transpile ?? transpile;
+        const transpileFn = init.transpile;
+        if (transpileFn === undefined) {
+          throw new Error(
+            "failed to generate .js/.d.ts files either the generators or the transpiler is required",
+          );
```

We could even do this check eagerly and fail fast.